### PR TITLE
Magiclysm - Golems

### PR DESCRIPTION
**Pull request by arijust**
#### Summary

```SUMMARY: Mods "Add golems to Magiclysm"```
#### Purpose of change

Introduces three golems:
Clay Golem, Stone Golem and Iron Golem.
While these can't be constructed by player, they still can be dismantled for resources they were made of.

#### Additional context

Temporary spawn point for these is forest.

### CODE REVIEW

File: data/mods/Magiclysm/monsters/monsters.json [hunk 1]
```diff
@@ -328,5 +328,100 @@
...
+    "special_attacks": [ { "type": "spell", "spell_id": "rocket_punch", "spell_level": 5, "cooldown": 10 } ],
+    "anger_triggers": [ "PLAYER_CLOSE", "HURT" ],
+    "death_function": [ "BROKEN" ],
+    "flags": [ "SEES", "NO_BREATHE", "ACID_IMMUNE", "LOUDMOVES" ]
+  },
+  {
+    "id": "mon_irongolem",
+    "type": "MONSTER",
+    "name": "iron golem",
+    "description": "Large human-like golem made from iron.  Sometimes, green gas is escaping its mouth.",
```
Comment (aoae23) [inline data/mods/Magiclysm/monsters/monsters.json]: I'd reword:
"A large, humanoid golem made from iron. Some sort of noxious gas seems to be seeping from its mouth."

File: data/mods/Magiclysm/recipes/deconstruction.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "components": [ [ [ "rock", 222 ] ], [ [ "sharp_rock", 33 ] ] ]
+  },
+  {
+    "result": "broken_irongolem",
+    "type": "uncraft",
+    "skill_used": "spellcraft",
+    "difficulty": 6,
+    "time": "6 h",
+    "qualities": [ { "id": "HAMMER", "level": 1 }, { "id": "SAW_M", "level": 1 } ],
+    "components": [ [ [ "scrap", 912 ] ] ]
```
Comment (aoae23) [inline data/mods/Magiclysm/recipes/deconstruction.json]: Would it also drop chunks and lumps of steel?

Comment (arijust) [reply data/mods/Magiclysm/recipes/deconstruction.json]: Sure.
I will lower number of scraps to add some lumps of steel.

File: data/mods/Magiclysm/Spells/monsterspells.json [hunk 3]
```diff
@@ -16,5 +16,44 @@
...
+    "description": "Ejects giant fist from arm.",
+    "valid_targets": [ "hostile" ],
+    "flags": [ "NO_LEGS" ],
+    "damage_type": "bash",
+    "min_damage": 30,
+    "max_damage": 40,
+    "damage_increment": 2,
+    "max_level": 5,
+    "min_range": 1,
+    "max_range": 2.5,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/monsterspells.json]: this value is incorrect as max_range is an integer. 2.5 will be truncated to 2, so you either need to make it 3 or 2.

File: data/mods/Magiclysm/Spells/monsterspells.json [hunk 4]
```diff
@@ -16,5 +16,44 @@
...
+    "damage_type": "bio",
+    "min_damage": 1,
+    "max_damage": 1,
+    "max_level": 5,
+    "min_range": 1,
+    "max_range": 3,
+    "range_increment": 0.2,
+    "effect": "target_attack",
+    "field_id": "fd_toxic_gas",
+    "field_chance": 100,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/monsterspells.json]: this spell confuses me: so you want a 1 in 100 chance of spawning  gas cloud on the tile the caster is standing on? what's the range for then? if you want to be able to spawn gas clouds on the ground you need to have `ground` as a valid target. additionally with an AOE of 0 it'll only affect the target tile (not necessarily a problem if that's the goal)

File: data/mods/Magiclysm/Spells/monsterspells.json [hunk 5]
```diff
@@ -16,5 +16,44 @@
...
+    "min_field_intensity": 1,
+    "max_field_intensity": 1,
+    "min_duration": 1000,
+    "max_duration": 6000
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/monsterspells.json]: there's no duration_increment

File: data/mods/Magiclysm/items/constructs.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    "type": "GENERIC",
+    "id": "broken_claygolem",
+    "symbol": "x",
+    "color": "brown",
+    "name": "broken clay golem",
+    "category": "other",
+    "description": "A broken clay golem, looking like piece of post-modern art.  Could be smashed for clay.",
+    "price": 10000,
+    "material": [ "clay" ],
+    "weight": "122000 g",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/constructs.json]:
```suggestion
    "weight": "122 kg",
```

File: data/mods/Magiclysm/items/constructs.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    "type": "GENERIC",
+    "id": "broken_stonegolem",
+    "symbol": "x",
+    "color": "gray",
+    "name": "broken stone golem",
+    "category": "other",
+    "description": "A broken stone golem, not that much different from big boulder.  Could be smashed for stone.",
+    "price": 10000,
+    "material": [ "stone" ],
+    "weight": "162000 g",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/constructs.json]:
```suggestion
    "weight": "162 kg",
```

File: data/mods/Magiclysm/items/constructs.json [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    "type": "GENERIC",
+    "id": "broken_irongolem",
+    "symbol": "x",
+    "color": "black",
+    "name": "broken iron golem",
+    "category": "other",
+    "description": "A broken iron golem, with all iron you would possibly ever need.  Could be smashed for iron.",
+    "price": 100000,
+    "material": [ "iron" ],
+    "weight": "945000 g",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/constructs.json]:
```suggestion
    "weight": "945 kg",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 9]
```diff
@@ -328,5 +328,100 @@
...
+    "id": "mon_claygolem",
+    "type": "MONSTER",
+    "name": "clay golem",
+    "description": "A large, humanoid golem made from clay.  Its proportions are off and it seems fragile.",
+    "default_faction": "robot",
+    "bodytype": "human",
+    "species": [ "ROBOT" ],
+    "diff": 2,
+    "volume": "120 L",
+    "weight": 122000,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json]:
```suggestion
    "weight": "122 kg",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 10]
```diff
@@ -328,5 +328,100 @@
...
+    "id": "mon_stonegolem",
+    "type": "MONSTER",
+    "name": "stone golem",
+    "description": "A large, humanoid golem made from stone.  Its fists look similar to rockets.",
+    "default_faction": "robot",
+    "bodytype": "human",
+    "species": [ "ROBOT" ],
+    "diff": 2,
+    "volume": "120 L",
+    "weight": 162000,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json]:
```suggestion
    "weight": "162 kg",
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 11]
```diff
@@ -328,5 +328,100 @@
...
+    "id": "mon_irongolem",
+    "type": "MONSTER",
+    "name": "iron golem",
+    "description": "A large, humanoid golem made from iron.  Some sort of noxious gas seems to be seeping from its mouth.",
+    "default_faction": "robot",
+    "bodytype": "human",
+    "species": [ "ROBOT" ],
+    "diff": 2,
+    "volume": "120 L",
+    "weight": 945000,
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json]:
```suggestion
    "weight": "945 kg",
```

File: data/mods/Magiclysm/Spells/monsterspells.json [hunk 12]
```diff
@@ -16,5 +16,48 @@
...
+    "max_range": 2,
+    "range_increment": 0.2,
+    "effect": "projectile_attack"
+  },
+  {
+    "id": "gas_attack",
+    "type": "SPELL",
+    "name": "Gas Attack",
+    "description": "Spreads toxic gas around itself.",
+    "valid_targets": [ "self" ],
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/monsterspells.json]:
```suggestion
    "valid_targets": [ "self", "hostile", "ground" ],
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 13]
```diff
@@ -328,7 +328,101 @@
     "harvest": "lemure",
     "anger_triggers": [ "PLAYER_CLOSE" ],
     "death_function": [ "NORMAL" ],
-    "regenerates": 1,
-    "flags": [ "SEES", "SMELLS", "HEARS", "STUMBLES", "PATH_AVOID_FIRE", "REVIVES" ]
+    "flags": [ "SEES", "SMELLS", "HEARS", "STUMBLES", "PATH_AVOID_FIRE", "REGENERATES_1", "REVIVES" ]
```
Comment (KorGgenT) [inline data/mods/Magiclysm/monsters/monsters.json]: this looks like a rebase error

Comment (arijust) [reply data/mods/Magiclysm/monsters/monsters.json]: It happens.

Comment (KorGgenT) [reply data/mods/Magiclysm/monsters/monsters.json]: image

Comment (arijust) [reply data/mods/Magiclysm/monsters/monsters.json]: I finally noticed that "regenerates" flag doesn't exist anymore.
